diag_manager/fms_diag_axis_object.F90

```diff
@@ -187,12 +193,14 @@ subroutine register_diag_axis_obj(this, axis_name, axis_data, units, cart_name,
 
     select type (axis_data)
     type is (real(kind=r8_kind))
-      allocate(real(kind=r8_kind) :: this%axis_data(size(axis_data)))
+      allocate(real(kind=r8_kind) :: this%axis_data(axis_length))
       this%axis_data = axis_data
+      this%length = axis_length
       this%type_of_data = "double" !< This is what fms2_io expects in the register_field call
     type is (real(kind=r4_kind))
-      allocate(real(kind=r4_kind) :: this%axis_data(size(axis_data)))
+      allocate(real(kind=r4_kind) :: this%axis_data(axis_length))
       this%axis_data = axis_data
+      this%length = axis_length
```

Comment:
This is optional. What happens if it's not included?

Reply:
Changed it to a required argument `fms_diag_axis_init` is only used locally here:
https://github.com/NOAA-GFDL/FMS/blob/cf55696892de0e5abec9914522b24778ff956a25/diag_manager/diag_axis.F90#L143-L145